### PR TITLE
[AIRFLOW-2936] Use official Python images as base image for Docker

### PR DESCRIPTION
**Pull request by jpds**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2936

### Description

- [X] Make Docker image use official Python images for base images.

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":

### CODE REVIEW

**Comment by jpds** [comment]
@dimberman

**Comment by jpds** [comment]
Difference in sizes:

```
airflow                                                latest                                                        79f38e11068e        About a minute ago   862MB
airflow-py2                                            latest                                                        4db3b3791c68        4 minutes ago        464MB
airflow-py3                                            latest                                                        c2739cc04bb1        9 minutes ago        465MB
```

**Comment by dimberman** [comment]
@jpds this looks great. Could you build the k8s executor from py3 instead of py2?

**Comment by jpds** [comment]
@dimberman Done.

File: scripts/ci/kubernetes/docker/airflow-init.sh [hunk 1]
```diff
@@ -17,9 +17,10 @@
...
-cd /usr/local/lib/python2.7/dist-packages/airflow && \
-cp -R example_dags/* /root/airflow/dags/ && \
+set -e
+
+cd /usr/local/lib/python3.7/site-packages/airflow/ && \
+cp -R example_dags/* /home/airflow/dags/ && \
 airflow initdb && \
 alembic upgrade heads && \
-(airflow create_user -u airflow -l airflow -f jon -e airflow@apache.org -r Admin -p airflow || true) && \
-echo "retrieved from mount" > /root/test_volume/test.txt
```
Comment (dimberman) [inline scripts/ci/kubernetes/docker/airflow-init.sh:25]: Why did you remove this line? Isn't this necessary for the volume mounting tests?

**Review by dimberman — CHANGES_REQUESTED** [review]

File: scripts/ci/kubernetes/docker/airflow-init.sh (hunk 1 above)
Comment (jpds) [reply scripts/ci/kubernetes/docker/airflow-init.sh:25]: I haven't found this necessary in our deployments (but we're using an Airflow image with our DAGs baked in).

Comment (dimberman) [reply scripts/ci/kubernetes/docker/airflow-init.sh:25]: So the reason this line didn't break the CI is because a previous PR silently broke all kubernetes tests. I'm actively trying to fix this and will report back when I have it working.
